src/runtime/particle-check.ts

```diff
@@ -20,14 +20,12 @@ export enum CheckType {
   IsFromStore = 'is-from-store',
 }
 
-export type CheckTarget = HandleConnectionSpec | ProvideSlotConnectionSpec;
-
-export class Check {
+export class Check implements CheckIntf {
   constructor(readonly target: CheckTarget, readonly expression: CheckExpression) {}
 
   toManifestString() {
     let targetString: string;
-    if (this.target instanceof HandleConnectionSpec) {
+    if (this.target.discriminator === 'HCS') {
```

Comment:
I think convention is to use `tag` and I'm not sure how type safe this is. This isn't blocking, just a note for revisiting.

Reply:
`tag` is already used in a slightly different way in other places, so I'd rather not overload it further. I found this disciminator pattern online, looked at a few versions, and found the most readable version to be the ones that called the discriminator `discriminator`. :-)

Reply:
In that case sounds good.